client/src/main/scala/org/apache/spark/delta/sharing/PreSignedUrlCache.scala

```diff
@@ -35,15 +35,18 @@ import io.delta.sharing.client.DeltaSharingProfileProvider
  *             remove the cached table from our cache.
  * @param lastAccess When the table was accessed last time. We will remove old tables that are not
  *                   accessed after `expireAfterAccessMs` milliseconds.
- * @param refresher the function to generate a new file id to pre sign url map, as long as the new
- *                  expiration timestamp of the urls.
+ * @param refresher the function to generate a new file id to pre sign url map, with the new
+ *                  expiration timestamp of the urls and the new refresh token.
+ * @param refreshToken the optional refresh token that can be used by the refresher to retrieve
+ *                     the same set of files with refreshed urls.
  */
 class CachedTable(
     val expiration: Long,
     val idToUrl: Map[String, String],
     val refs: Seq[WeakReference[AnyRef]],
     @volatile var lastAccess: Long,
-    val refresher: () => (Map[String, String], Option[Long]))
+    val refresher: Option[String] => (Map[String, String], Option[Long], Option[String]),
```

Comment:
I should do it in previous change: Instead of returning a 3 elements tuple, let's use a case class and return the class from refresher function, I don't have a good name, maybe `RefreshedResult`/`RefreshedUrlsAndMetadata`?

Reply:
Done!